controllers/storagecluster/generate.go

```diff
@@ -83,3 +90,28 @@ func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string)
 func generateStorageQuotaName(storageClassName, quotaName string) string {
 	return fmt.Sprintf("ocs-%s-%s", storageClassName, quotaName)
 }
+
+// generateCephFSProviderParameters function generates extra parameters required for provider storage clusters
+func generateCephFSProviderParameters(initData *ocsv1.StorageCluster) map[string]string {
+	if initData.Spec.AllowRemoteStorageConsumers {
+		deviceSetList := initData.Spec.StorageDeviceSets
+		var deviceSet ocsv1.StorageDeviceSet
+		for _, ds := range deviceSetList {
+			if ds.Name == StorageDeviceName {
+				deviceSet = ds
+			}
+		}
+		deviceCount := deviceSet.Count
+		pgNumValue := DefaultPGnumValue
+		if deviceCount > DefaultDCFor4TB {
+			pgNumValue = ProviderPgNumValue
+		}
```

Comment:
This is not the logic we agreed to! this will only work for 4Tib and 20Tib! what will happen with 8Tib, 16Tib etc?

You should calculate the total num of PG available based on the device count from that get a value for a single unit, the ont that we will use for block pools, and set the CephFS block pool to use 3 such units.

I would suggest moving the calculation of the `unit size` into some other location as you will need to reuse it in the storage consumer controller when creating the block pool per consumer.

Reply:
Have changed implementation as discussed